### ISSUE TITLE
Oddity - import doesn't complete, if a new one is started in another tab weird stuff happens

### ISSUE THREAD

**Issue by RCheesley** [mautic/mautic#74]
I imported leads from several sources using the CSV import - everything seems to be OK (although it always seems to stop ~7-9 entries before the end of the import).  I can't see any reason why, but with a ~4k import it stalled with 9 to go, with a ~2.5k import it stalled with 7 to go.

I had accidentally left an import window open in another tab to run for several hours and forgotten (the one with 4k to import, stuck with 9 leads to go), when I started a subsequent import in a new tab, the old import started up again, showing the exact same progress as the new import - seemed a bit weird, not sure if this is a bug or just a quirk (or indeed how it's supposed to work).

**Comment by alanhartless** [comment]
The import option does make use of sessions (it stores the progress and headers in the session) so you will see funky things if you try to do an import in multiple tabs of the same browser.  If you accidentally close the tab, it should pick up where left off when returning unless the session is lost.

Are you able to determine if those 7-9 entries are definitely missing or is it that they did get imported but  the imported doesn't "realize" it's finished?

Thanks!
Alan

**Comment by RCheesley** [comment]
I think it's the latter, every single time I run an import, there's around 7 left when it stops.

**Comment by alanhartless** [comment]
Hey Ruth,

If you open those CSV files in a text editor, are there by chance blank lines at the bottom?  I ran tests with fake leads generated from http://www.fakenamegenerator.com/order.php (great tool) where I had a list of 500, 5000, 10000, and then tried a couple where I trimmed off a bunch to get a random number and in all instances, the importer completed with the expected numbers matching.

I'm just wondering if there is something about your specific csv files that could be leading to different results?  What software did you export the results from?

Thanks!
Alan
